### PR TITLE
feat: manage remote seller blacklist

### PR DESCRIPTION
**Pull request by levalleux-ludo**
- replace env var REACT_APP_SELLER_CURATION_LIST with REACT_APP_SELLER_BLACKLIST_URL, so that the seller curation list is an external file that can be modified without rebuilding/redeploying the dApp
- ensure the seller curation list is enabled by default (and default value is [] = no sellers, in case the seller blacklist file can't be fetched)
- ensure the "custom storefront" seller list can restrict the curated seller list, but can not extend it (meaning that custom storefront can only show a subset of whitelisted sellers)
- ensure the "seller profile" page is only available for curated sellers
- ensure the "explore" pages only shows lists of offers/products/sellers of the curated seller list
- ensure the "about" page shows the effective list of curated sellers
- ensure no offer/product/exchange/dispute/chatChannel is no longer shown if the seller is blacklisted
- ensure a blacklisted seller can't access the seller dashboard, custom storefront and create product pages anymore

Blacklist have been prepared in this repo, for each env: https://github.com/BAppLimited/curationLists/blob/main/bosonApp.io/testing/sellers/blacklist.json
https://github.com/BAppLimited/curationLists/blob/main/bosonApp.io/staging/sellers/blacklist.json
https://github.com/BAppLimited/curationLists/blob/main/bosonApp.io/production/sellers/blacklist.json

### CODE REVIEW

File: src/lib/utils/hooks/useCurationLists.ts [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
-    enableCurationLists:
-      withOwnProducts === "all"
-        ? false
-        : ["mine", "custom"].includes(withOwnProducts)
-        ? true
-        : CONFIG.enableCurationLists,
-    sellerCurationList: sellerCurationListFromUrl
-      ? parseCurationList(sellerCurationListFromUrl)
-      : CONFIG.sellerCurationList,
+    enableCurationLists: CONFIG.enableCurationLists, // curation list is always enabled except if env var is used to disable it
```
Comment (albertfolch-redeemeum) [inline src/lib/utils/hooks/useCurationLists.ts:23]: I think this comment would make sense if it was hardcoded to true? so please remove the comment if my understanding is correct

File: src/lib/utils/hooks/useSellerBlacklist.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    {
+      ...options
+    }
+  );
+  const blacklist = useQuery(
+    "seller-blacklist",
+    async () => {
+      const blacklistStr = props.sellerBlacklistUrl
+        ? await fetchTextFile(props.sellerBlacklistUrl, false)
```
Comment (albertfolch-redeemeum) [inline src/lib/utils/hooks/useSellerBlacklist.ts:55]: should we catch this exception if this fails and then throw it so we can capture it with Sentry and alert us if the json file we use for blacklisting cannot be fetched?

File: src/lib/utils/hooks/useSellerWhitelist.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    {
+      ...options
+    }
+  );
+  const blacklist = useQuery(
+    "seller-blacklist",
+    async () => {
+      const blacklistStr = props.sellerBlacklistUrl
+        ? await fetchTextFile(props.sellerBlacklistUrl, false)
```
Comment (albertfolch-redeemeum) [inline src/lib/utils/hooks/useSellerWhitelist.ts:41]: same question as above

File: src/lib/utils/hooks/useSellerWhitelist.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        : "";
+      try {
+        return JSON.parse(blacklistStr).map((id: unknown) =>
+          String(id)
+        ) as string[];
+      } catch (e) {
+        console.error(e);
```
Comment (albertfolch-redeemeum) [inline src/lib/utils/hooks/useSellerWhitelist.ts:48]: catch error with Sentry

File: src/lib/utils/textFile.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export async function fetchTextFile(textFileUrl: string, useCache = true) {
+  const response = await fetch(
+    textFileUrl,
+    useCache ? {} : { cache: "no-cache" }
+  );
+  if (!response.ok) {
+    throw `Error when fetching ${textFileUrl}: ${response.status} ${response.statusText}`;
```
Comment (albertfolch-redeemeum) [inline src/lib/utils/textFile.ts:7]: maybe we can catch it here instead (question above)

File: src/pages/sell/SellerCenter.tsx [hunk 6]
```diff
@@ -136,6 +143,10 @@ function SellerCenterWrapper() {
     </select>;
   }
 
+  if (!selectedSellerId) {
+    return <div data-testid="notCuratedSeller">Not found Seller.</div>;
```
Comment (albertfolch-redeemeum) [inline src/pages/sell/SellerCenter.tsx:147]:
```suggestion
    return <div data-testid="notCuratedSeller">Seller not found.</div>;
```

**Review by albertfolch-redeemeum — APPROVED** [review]
minor questions/comments but looks good!

File: src/lib/utils/hooks/useSellerBlacklist.ts (hunk 2 above)
Comment (levalleux-ludo) [reply src/lib/utils/hooks/useSellerBlacklist.ts:55]: Ok, to capture the error with Sentry. Will do it in fetchTextFile below.

Comment (levalleux-ludo) [reply src/lib/utils/hooks/useSellerBlacklist.ts:55]: Here my intention is to let the exception throwing to ensure `useQuery()` is failing (retrying N times?, then returning `isError=true`)

File: src/lib/utils/hooks/useSellerWhitelist.ts (hunk 3 above)
Comment (levalleux-ludo) [reply src/lib/utils/hooks/useSellerWhitelist.ts:41]: same answer :-)

File: src/lib/utils/hooks/useSellerWhitelist.ts (hunk 4 above)
Comment (levalleux-ludo) [reply src/lib/utils/hooks/useSellerWhitelist.ts:48]: sentry capture added

File: src/lib/utils/textFile.ts (hunk 5 above)
Comment (levalleux-ludo) [reply src/lib/utils/textFile.ts:7]: sentry capture added

File: src/lib/utils/hooks/useExchanges.ts [hunk 7]
```diff
@@ -56,6 +59,7 @@ const getExchangesFunction = async (props: Props) => {
       $offerId: String
       ${first ? "$first: Int" : ""}
       ${skip ? "$skip: Int" : ""}
+      ${seller_in ? "$seller_in: String" : ""}
```
[the remark this answers is not in the document]
Comment (levalleux-ludo) [reply src/lib/utils/hooks/useExchanges.ts:62]: Btw it's a mistake to add seller_in as a gql variable here (as it's never referenced as is). 
Setting it mandatory leads to an error.
So I prefer to remove the line

**Pull request merged**